flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/typeutils/BackwardsCompatibleAvroSerializerTest.java

```diff
@@ -26,7 +26,7 @@
 import org.apache.flink.api.java.typeutils.runtime.PojoSerializer;
 import org.apache.flink.api.java.typeutils.runtime.PojoSerializer.PojoSerializerConfigSnapshot;
 import org.apache.flink.core.memory.DataInputViewStreamWrapper;
-import org.apache.flink.formats.avro.generated.User;
+import org.apache.flink.formats.avro.generated.SimpleUser;
```

Comment:
The problem is that {{BackwardsCompatibleAvroSerializer}} does not support records with logical types. Logical types need a Kryo configuration that the serializer does not set correctly. This might be a bug or at least a missing feature. Given that this serializer only exists for backwards compatibility for 1.3 (which used Avro 1.7 without logical type), I added a simple user for this test. I will add a comment about this to the code.